Add driver type information in service-client telemetry

sonalideshpandemsft opened the pull request:
This change adds `driver`, `driverEndpointName` and `testVariant` information to service-client telemetry. Rest of the field such as `clientType`, `loaderId` and `containerId` are not displayed as part of `Test_start` and `Test_end` event name

[ADO Work Item](https://dev.azure.com/fluidframework/internal/_workitems/edit/8762)

sonalideshpandemsft committed:
log data

Josmithr committed:
build(docs): Hard-code redirects (#22949)

Updates website build to no longer dynamically generate
`fallback/index.js`. Instead, the redirects are hard-coded in
`fallback/routes.js`.

This work is in preparation for the Docusaurus migration.


[AB#19990](https://dev.azure.com/fluidframework/235294da-091d-4c29-84fc-cdfc3d90890b/_workitems/edit/19990)

sonalideshpandemsft committed:
Merge branch 'log-data' of https://github.com/sonalideshpandemsft/FluidFramework into test/log-data

Abe27342 committed:
Improve DDS stress pipeline stage rendering (#22952)

## Description

Adjusts the `condition` parameter of
`include-conditionally-run-stress-test` to apply to the entire stage
rather than the test-running job. This improves the clarity from the
pipeline overview page, see most recent run vs previous ones:


![image](https://github.com/user-attachments/assets/e344d02a-a617-4163-bbda-0bfaac276541)

This also fixes the `stressMode` parameter not being given options at
queue time, which was introduced in #21702 and prevented the pipeline
from being queued.

---------

Co-authored-by: Abram Sanderson <absander@microsoft.com>

CraigMacomber committed:
Tree: allow explicit construction of array and map nodes with no arguments (#22946)

## Description

See changeset for details.

noencke committed:
Fix SharedTreeCore detached sequence numbers when loading into a detached tree (#22947)

## Description

It is possible to load a summary into a detached SharedTree. In this
case, the detached revision that is used to generate sequence IDs for
commits while detached must be updated to ensure that it doesn't
duplicate any of the sequence IDs already used in the summary. This PR
fixes the issue and also adds an assert when sequencing in the
EditManager to ensure that we don't sequence regressive sequence IDs.

This fix also allows us to trim the trunk when summarizing (without
breaking our fuzz tests), which reduces summary sizes especially for
detached trees with many synchronous edits.

Abe27342 commented on packages/service-clients/end-to-end-tests/azure-client/src/test/.mocharc.cjs:
```diff
@@ -8,4 +8,24 @@
+	const testVariant = getFluidTestVariant();
+	process.env.FLUID_TEST_VARIANT = testVariant;
+
+	return config(packageDir, additionalRequiredModules, testVariant);
```
this looks wrong, seems like you want `getFluidTestMochaConfig(packageDir, additionalRequiredModules, testVariant)` here. Same in other mocharc

sonalideshpandemsft replied:
thanks for catching that!

Abe27342 commented on packages/service-clients/end-to-end-tests/odsp-client/package.json:
```diff
@@ -30,7 +30,7 @@
 		"test:coverage": "c8 npm test",
 		"test:realsvc:odsp": "cross-env npm run test:realsvc:odsp:run",
-		"test:realsvc:odsp:run": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit --timeout 20000",
+		"test:realsvc:odsp:run": "mocha --recursive \"lib/test/**/*.spec.*js\" --exit --timeout 20000 --config src/test/.mocharc.cjs -- --driver=odsp --driverEndpoint=odsp",
```
Is there a reason for the `--` (i.e. not passing `--driver=odsp` directly to mocha)?

AFAIK that syntax is mostly just a thing for npm, where generally speaking, command-line arguments to `npm` are not plumbed through to child processes it creates. So e.g. `npm run foo --someFlag` runs the script `foo` and passes `someFlag` to the `npm` process, but whatever code executes as a result of running the `foo` script won't see `--someFlag` in its arguments. However `npm run foo -- --someFlag` is the opposite: the npm process doesn't have `--someFlag` but the script for `foo` will.

sonalideshpandemsft committed:
Correct function name

CraigMacomber committed:
tree: Make SchematizingSimpleTreeView implement TreeViewAlpha (#22953)

## Description

Make SchematizingSimpleTreeView implement TreeViewAlpha, and add some
helper types.

sonalideshpandemsft committed:
fix function

sonalideshpandemsft committed:
Fix mocharc

zhenmichael committed:
(docs): update force deployment condition for build docs (#22954)

previous shouldDeploy condition have not been included in deployment
condition. This variable has been updated to "forceDeployment" and
included back in the deployment condition.

shouldDeploy has been updated to check if release builds are the latest
version in its major version series.

sonalideshpandemsft committed:
Merge branch 'test/log-data', remote-tracking branch 'origin' into log-data

sonalideshpandemsft committed:
Merge branch 'log-data' of https://github.com/sonalideshpandemsft/FluidFramework into log-data

sonalideshpandemsft commented:
closing in favor of https://github.com/microsoft/FluidFramework/pull/22964
